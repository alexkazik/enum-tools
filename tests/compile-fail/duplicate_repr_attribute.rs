#![no_implicit_prelude]

use ::enum_tools::EnumTools;

#[derive(EnumTools, Debug, Clone, Copy, Eq, PartialEq)]
#[repr(i8)]
#[repr(i8)]
pub(crate) enum EG {
    #[enum_tools(rename = "A*")]
    A,
    B,
    C,
    D,
}

fn main() {}
