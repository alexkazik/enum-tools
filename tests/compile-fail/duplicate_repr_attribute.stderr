error: duplicate repr attribute
 --> tests/compile-fail/duplicate_repr_attribute.rs:6:8
  |
6 | #[repr(i8)]
  |        ^^

error[E0566]: conflicting representation hints
 --> tests/compile-fail/duplicate_repr_attribute.rs:6:8
  |
6 | #[repr(i8)]
  |        ^^
7 | #[repr(i8)]
  |        ^^
  |
  = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #68585 <https://github.com/rust-lang/rust/issues/68585>
  = note: `#[deny(conflicting_repr_hints)]` on by default
